test/quantization/test_quantized_op.py

```diff
@@ -2274,6 +2274,38 @@ def test_empty_batch(self):
         result = torch.ops.quantized.linear_dynamic(X, w_packed)
         self.assertEqual(result.shape, (0, 2))
 
+    def test_advanced_indexing(self):
+        """
+        Verifies that the x[:, [0], :, :] syntax works for quantized tensors.
+        """
+        for dtype in (torch.qint8, torch.quint8, torch.qint32):
+            x_q = torch.quantize_per_tensor(
+                torch.randn(1, 4, 4, 4), 0.1, 0, dtype)
+            # reference
+            x_fp32 = x_q.dequantize()
+
+            # single dim, single index
+            x_q_s1 = x_q[:, [0], :, :]
+            x_fp32_s1 = x_fp32[:, [0], :, :]
+            self.assertTrue(torch.allclose(x_fp32_s1, x_q_s1.dequantize()))
+            self.assertTrue(x_q.dtype == x_q_s1.dtype)
+            self.assertTrue(x_q.q_scale() == x_q_s1.q_scale())
+            self.assertTrue(x_q.q_zero_point() == x_q_s1.q_zero_point())
+
+            # multiple dim, single index
+            x_q_s2 = x_q[:, [0], [2], :]
+            x_fp32_s2 = x_fp32[:, [0], [2], :]
+            self.assertTrue(torch.allclose(x_fp32_s2, x_q_s2.dequantize()))
```

Comment:
Might it be useful to check that the int_repr, scale and zero-point are identical instead of the fp32 check.

Reply:
sure, added an additional check for this for the simple case